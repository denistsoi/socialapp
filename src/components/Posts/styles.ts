import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  postContainer: {
    padding: 2,
    paddingBottom: 3,
    borderBottomColor: "#ababab",
    borderBottomWidth: 1,
    marginBottom: 8
  },
  postContent: {
    marginBottom: 4
  },
  postTitle: {
    fontWeight: "bold",
    color: "#666"
  },
  postBody: {
    fontSize: 10,
    color: "#888"
  },
  commentButton: {
    alignSelf: "flex-end",
    backgroundColor: "#efefef",
    padding: 4
  },
  comment: {
    fontSize: 10
  }
});

export default styles;
