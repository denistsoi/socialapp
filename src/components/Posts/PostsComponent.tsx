import React, { ReactElement } from "react";
import { View, Text } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";
import { Post } from "../../types/resource";

import styles from "./styles";

const Posts = ({
  posts,
  onPress
}: {
  posts: Array<Post>;
  onPress: Function;
}): ReactElement => {
  return (
    <View>
      {posts?.map((post, index) => (
        <View key={index} style={styles.postContainer}>
          <View style={styles.postContent}>
            <Text numberOfLines={1} style={styles.postTitle}>
              {post.title}
            </Text>
            <Text numberOfLines={2} style={styles.postBody}>
              {post.body}
            </Text>
          </View>
          <View style={styles.commentButton}>
            <TouchableOpacity onPress={(): void => onPress(post)}>
              <Text style={styles.comment}>Comments</Text>
            </TouchableOpacity>
          </View>
        </View>
      ))}
    </View>
  );
};

export default Posts;
