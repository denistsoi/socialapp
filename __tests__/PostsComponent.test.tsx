import React from "react";
import { render } from "react-native-testing-library";
import Posts from "../src/components/Posts/PostsComponent";

describe("Posts Component", () => {
  it("should render single data in Posts", async () => {
    let props: any;

    props = {
      onPress: jest.fn(),
      posts: [
        {
          id: 1,
          userId: 1,
          title: "Bob cat",
          body: "I am your number #1 fan Bob!"
        }
      ]
    };

    const component = render(
      <Posts onPress={props.onPress} posts={props.posts} />
    );

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.getByText("Bob cat")).toBeDefined();
    expect(component.getByText("I am your number #1 fan Bob!")).toBeDefined();
  });
});
