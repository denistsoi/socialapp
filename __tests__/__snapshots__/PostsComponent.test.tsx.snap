// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Component should render single data in Posts 1`] = `
<View>
  <View
    style={
      Object {
        "borderBottomColor": "#ababab",
        "borderBottomWidth": 1,
        "marginBottom": 8,
        "padding": 2,
        "paddingBottom": 3,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 4,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#666",
            "fontWeight": "bold",
          }
        }
      >
        Bob cat
      </Text>
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#888",
            "fontSize": 10,
          }
        }
      >
        I am your number #1 fan Bob!
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#efefef",
          "padding": 4,
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 10,
              }
            }
          >
            Comments
          </Text>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
</View>
`;
